ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryController.java

```diff
@@ -355,13 +375,34 @@ public void onMessage(final Packet packet, final DiscoveryPeer sender) {
         }
         break;
       case ENR_RESPONSE:
-        // Currently there is no use case where an ENRResponse will be sent otherwise
-        // logic can be added here to query and store the response ENRs
-        packet
-            .getPacketData(ENRResponsePacketData.class)
-            .filter(p -> p.getEnr().getNodeId().equals(sender.getId()))
-            .ifPresent(p -> LOG.debug("Received NodeRecord: {}", p.getEnr().asEnr()));
-
+        matchInteraction(packet)
+            .ifPresent(
+                interaction -> {
+                  final Optional<ENRResponsePacketData> packetData =
+                      packet.getPacketData(ENRResponsePacketData.class);
+                  final NodeRecord enr = packetData.get().getEnr();
+                  peer.setNodeRecord(enr);
+
+                  Optional<ForkId> maybeForkId = peer.getForkId();
+                  if (maybeForkId.isPresent()) {
```

Comment:
does this checking only happen if the --Xfilter-on-enr-fork-id is set?

Reply:
Yes, if we do not want to filter on the forkid we do not send an ENR request, which means that the peer does not send the ENR response.